enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+- In order to use ImagePolicy, an existing cluster will have to make an upgrade to the version that the `ImagePolicy` CRD is available.
+- If the existing cluster manages `/etc/containers/policy.json` via the MachineConfig, the configuration will continue to work if no new `ImagePolicy` resources created during the upgrade. If `ImagePolicy` resources created and resulting `/path/to/policies/\<NAMESPACE\>.json` exists, CRI-O will use the config from `/path/to/policies/\<NAMESPACE\>.json`.
+
+Downgrade expectations: 
+- Not applicable. 
+  If `N` does not support ImagePolicy CRD, `N+1`supports IamgePolicy. It is not expected that the ImagePolicy related to the failure `N->N+1`. New resources should not be created during `N->N+1`.
+
+
+### Version Skew Strategy
+
+The MCO does not require skew check.
+The `ImagePolicy` involve coordinating behavior in the CRI-O v1.28. The node team will not expose the `ImagePolicy` CRD if the CRI-O/MCO does not support the namespaced signature to avoid the version skew with CRI-O.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+- CR field syntax error failure reported from CLI
+- CR conflicting value failure: rollout failure according to the merge rules [Implementation Details](#Update-container-runtime-config-controller-to-watch-ImagePolicy). The failure will be reported by machine config controller logs
+- MCO rolling out configuration file failure reported by MCO
+- If the signature validation fails, then CRI-O will report that in the same way as for any other pull failure. Further enhancements to the kubelet, CRI and CRI-O error handling can be achieved in future Kubernetes releases.
+
+The above errors should not impact the overall cluster health
+OCP Node team is likely to be called upon in case of escalation with one of the failure modes.
+
+#### Support Procedures
+
+- detect the failure modes in a support situation, describe possible symptoms (events, metrics,
+  alerts, which log output in which component)
+  - If the ImagePolicy CR does not follow the syntax requirements, api-server will fail when creating the objects.
+  - If the ImagePolicy CR is not rolled out by MCO, machine-config-controller logs will shows error with the reason.
+
+- disable the API extension (e.g. remove MutatingWebhookConfiguration `xyz`, remove APIService `foo`)
+  - Disabling/removing the CRD is not possible without removing the CR instances. Customer will lose data.
+
+- What consequences does it have on existing, running workloads?
+  - No impact. CRI-O does not verify signature verification for running containers.
```

Comment:
To roll this out, will MCO reboot the hosts? If so, there's an impact since the containers will all be restarted

Reply:
It does not need the reboot, we don't want to ask for the reboot every time a namespace admin makes the change.  follows the current [Reload Crio](https://github.com/openshift/machine-config-operator/blob/ff7ef2ec8ddbdf4f5758ee8f3ba3fea2d364e581/docs/MachineConfigDaemon.md?plain=1#L165) action.

Reply:
Might be worth clarifying that within the doc here